rclcpp/src/rclcpp/clock.cpp

```diff
@@ -118,18 +137,30 @@ Clock::create_jump_callback(
     throw std::bad_alloc{};
   }
 
-  // Try to add the jump callback to the clock
-  rcl_ret_t ret = rcl_clock_add_jump_callback(
-    &rcl_clock_, threshold, Clock::on_time_jump,
-    handler.get());
-  if (RCL_RET_OK != ret) {
-    exceptions::throw_from_rcl_error(ret, "Failed to add time jump callback");
+  // Lookup the per-object mutex
+  g_clock_map_mutex.lock_shared();
+  std::mutex & clock_mutex = g_clock_mutex_map.at(this);
+  g_clock_map_mutex.unlock_shared();
+  {
+    std::lock_guard<std::mutex> clock_guard(clock_mutex);
+    // Try to add the jump callback to the clock
+    rcl_ret_t ret = rcl_clock_add_jump_callback(
+      &rcl_clock_, threshold, Clock::on_time_jump,
+      handler.get());
+    if (RCL_RET_OK != ret) {
+      exceptions::throw_from_rcl_error(ret, "Failed to add time jump callback");
+    }
   }
 
   // *INDENT-OFF*
   // create shared_ptr that removes the callback automatically when all copies are destructed
   // TODO(dorezyuk) UB, if the clock leaves scope before the JumpHandler
   return JumpHandler::SharedPtr(handler.release(), [this](JumpHandler * handler) noexcept {
+    g_clock_map_mutex.lock_shared();
+    std::mutex & clock_mutex = g_clock_mutex_map.at(this);
+    g_clock_map_mutex.unlock_shared();
```

Comment:
In fact, as often as you do this, it might be nice to have a function for it, like `get_clock_mutex(Clock *) -> std::mutex &`.

Reply:
Good point.  I'll go ahead and make a function for it, which should solve both of your concerns.